source/image-handler/test/test-image-request.js

```diff
@@ -641,6 +641,25 @@ describe('decodeRequest()', function() {
             });
         });
     });
+    describe('004/signedPath', function() {
+        it(`Should throw an error if signature is invalid`, function() {
+            // Arrange
+            process.env = {
+                SIGNATURE_KEY : "mySecretKey"
+            }
+            const path = 'eyJlZGl0cyI6eyJncmF5c2NhbGUiOiJ0cnVlIiwicm90YXRlIjo5MCwiZmxpcCI6InRydWUifX0--invalid-signature'
+            // Act
+            const imageRequest = new ImageRequest();
+            // Assert
+            assert.throws(function() {
+                imageRequest.decodeRequest(event);
+            }, Object, {
+                status: 400,
+                code: 'DecodeRequest::CannotReadPath',
+                message: 'The URL path you provided could not be read. Please ensure that it is properly formed according to the solution documentation.'
```

Comment:
My reading of the diff makes this error message surprising. I didn't read the whole thing though.